### PR TITLE
Refactor Build & Set Hostname for Switch Inventory

### LINKED ISSUE #188: Speed up Setting Hostnames
Need to explore options (API & path queries) for speeding up setting the hostname tasks.
Ideally reduce Ansible tasks and move into native Python.

### PR DESCRIPTION
**Pull request by mtarking**
Resolves #188 
Refactor build & set hostname for switch inventory using a Python plugin to speed up execution in larger environments.

### CODE REVIEW

File: plugins/action/dtc/build_hostname_policy_payload.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                task_vars=task_vars,
+                tmp=tmp
+            )
+            policy_match = next((item for item in policy_data["response"]["DATA"] if item["templateName"] == "host_11_1"))
+            if policy_match["nvPairs"]["SWITCH_NAME"] != switch["name"]:
+                policy_match["nvPairs"]["SWITCH_NAME"] = switch["name"]
+                policy_payload.append(policy_match)
+
+        results['policy_payload'] = policy_payload
+        results['policy_ids'] = "%2C".join([str(policy["id"]) for policy in policy_payload])
```
Comment (dsx1123) [inline plugins/action/dtc/build_hostname_policy_payload.py:55]: suggest to return list instead of comma separated string, join the list in the playbook

**Review by dsx1123 — CHANGES_REQUESTED** [review]

File: plugins/action/dtc/build_hostname_policy_payload.py (hunk 1 above)
Comment (mtarking) [reply plugins/action/dtc/build_hostname_policy_payload.py]: returned just a list and re-added a task to perform the conversion to string list.
also added a result changed knob.

File: plugins/action/dtc/build_hostname_policy_payload.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from ansible.plugins.action import ActionBase
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        results = super(ActionModule, self).run(tmp, task_vars)
+        results['changed'] = False
+
+        model_data = self._task.args["model_data"]
```
Comment (dsx1123) [inline plugins/action/dtc/build_hostname_policy_payload.py:36]: @mtarking after second thought, I feel like we might need similar function in the future to update other policies that are created by the NDFC, will it be a good idea to make it generic? The input should be a list of switch serial numbers and the output is a list of dict:
```json
{
   "{serial_number}": "POLICY-XXXXX",
}
```

Comment (mtarking) [reply plugins/action/dtc/build_hostname_policy_payload.py]: Discussed verbally and agreed.

**Review by dsx1123 — APPROVED** [review]

**Pull request merged**